neqo-http3/src/connection.rs

```diff
@@ -1047,22 +1047,25 @@ mod tests {
     }
 
     // Start a client/server and check setting frame.
+    #[allow(clippy::cognitive_complexity)]
     fn connect(client: bool) -> (Http3Connection, Connection) {
         // Create a client/server and connect it to a server/client.
         // We will have a http3 server/client on one side and a neqo_transport
         // connection on the other side so that we can check what the http3
         // side sends and also to simulate an incorrectly behaving http3
         // server/client.
 
-        let mut hconn;
-        let mut neqo_trans_conn;
-        if client {
-            hconn = Http3Connection::new(default_client(), 100, 100, None);
-            neqo_trans_conn = default_server();
+        let (mut hconn, mut neqo_trans_conn) = if client {
```

Comment:
I don't like how clippy warns about this one.  But I'm not in a mood to suppress the change, even if it isn't great.